File: templates/problems/18_bit_manipulation/05_missing_number/typescript/missingNumber.test.ts
import { describe, it, expect } from 'vitest';
import { missingNumber } from './missingNumber';

describe('MissingNumber', () => {
    it('missing 2', () => {
        expect(missingNumber([3, 0, 1])).toBe(2);
    });

    it('missing 2 from 3', () => {
        expect(missingNumber([0, 1])).toBe(2);
    });

    it('missing 8', () => {
        expect(missingNumber([9, 6, 4, 2, 3, 5, 7, 0, 1])).toBe(8);
    });

    it('missing 0', () => {
        expect(missingNumber([1])).toBe(0);
    });

    it('missing 1', () => {
        expect(missingNumber([0])).toBe(1);
    });

    it('single zero', () => {
        expect(missingNumber([0, 2, 3])).toBe(1);
    });
});

File: templates/problems/18_bit_manipulation/05_missing_number/go/missing_number_test.go
package missing_number

import "testing"

func TestMissingNumber(t *testing.T) {
	tests := []struct {
		name     string
		nums     []int
		expected int
	}{
		{"missing 2", []int{3, 0, 1}, 2},
		{"missing 2 from 3", []int{0, 1}, 2},
		{"missing 8", []int{9, 6, 4, 2, 3, 5, 7, 0, 1}, 8},
		{"missing 0", []int{1}, 0},
		{"missing 1", []int{0}, 1},
		{"single zero", []int{0, 2, 3}, 1},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := missingNumber(tt.nums)
			if got != tt.expected {
				t.Errorf("missingNumber(%v) = %d, want %d", tt.nums, got, tt.expected)
			}
		})
	}
}
